teek/_widgets/treeview.py

```diff
@@ -0,0 +1,545 @@
+                        set_config=False)
+            rows.append(row)
+
+        return rows
+
+    @make_thread_safe
+    def insert(self, index, row):
+        """
+        Insert a row or values converted to a row at the given index.
+        """
+        # Convert to row
+        if not isinstance(row, TreeviewRow):
+            row = TreeviewRow(values=row)
+
+        self._treeview._call(
+            None, self._treeview, 'insert', self._root, index, '-id', row
+        )
+
+        row._assign(treeview=self._treeview, root=self._root)
+
+    @make_thread_safe
+    def sort(self, column_pos, reverse=False):
+        """
+        Sort all rows according to value of column with given ``column_pos``.
+        The direction of the sorting can be set with ``reverse``. If this value
+        is true, the rows are sorted in ascending order.
+        """
+        if column_pos < 1:
+            raise IndexError('column position must be greater or equal 1')
+
+        sorted_rows = list(self).copy()
+        sorted_rows.sort(key=lambda r: r.config['values'][column_pos - 1],
+                         reverse=reverse)
+
+        for to_pos, row in enumerate(sorted_rows):
+            from_pos = self.index(row)
+
+            if from_pos != to_pos:
+                row.move(to_pos)
+
+            row.subrows.sort(column_pos, reverse)
+
+
+class TreeviewRow:
+    """
+    Represents a row of a treeview widget or one which is ready to be added
+    to a treeview widget. The row holds all the column values and additional
+    options like title text or icon. Like a :class:`TreeviewColumn`, options
+    can only be get/set or methods can be executed if the :class:`TreeviewRow`
+    is connected to a :class:`Treeview`.
+
+    If you create a :class:`TreeviewRow` instance, it is not initially
+    connected to a :class:`Treeview`. This is automatically done when it is
+    appended to the *row* list of the :class:`Treeview` instance like this:
+    ::
+
+        treeview.rows.append(teek.TreeviewRow())
+
+    A row can be created without any arguments. Then the name of the row
+    is automatically generated. If the first argument is given, then this name
+    is taken.
+
+    For convenience, a column can also be created using...
+    ::
+
+        treeview.rows.append(['a', 'b', 'c'])
+
+    ...which is exactly the same as this:
+    ::
+
+        treeview.rows.append(teek.TreeviewRow(values=['a', 'b', 'c']))
+
+    There are never multiple :class:`TreeviewRow` objects that represent the
+    same row.
+
+    :class:`TreeviewRow` objects can also contain nested rows (subrows). They
+    are created like normal rows but with the `.subrows` attribute:
+    ::
+
+        treeview.rows[0].subrows.append(teek.TreeviewRow(values=['a', 'b'])
+
+    .. attribute:: config
+
+        Similar to the ``config`` attribute that widgets have. The available
+        options are documented as ``ITEM OPTIONS`` in :man:`ttk_treeview(3tk)`.
+    """
+    _next_row_num = 0
```

Comment:
Should this really be global to the whole program instead of being specific to a treeview instance?